src/array_api_extra/_lib/_compat.pyi

```diff
@@ -11,3 +11,6 @@ def array_namespace(
     use_compat: bool | None = None,
 ) -> ArrayModule: ...
 def device(x: Array, /) -> Device: ...
+def is_array_api_obj(x: object, /) -> bool: ...
+def is_dask_array(x: object, /) -> bool: ...
+def is_writeable_array(x: object, /) -> bool: ...
```

Comment:
Maybe a `TypeGuard` could be used here.
I'm not sure if at this point it'll actually help, but it should at least be valid 🤷🏻

Reply:
TypeGuard is a very good idea, but I'd rather invest the energy to implement it in `array_api_compat`